python/ray/data/datasource/file_based_datasource.py

```diff
@@ -495,13 +496,20 @@ def _resolve_paths_and_filesystem(
         )
         try:
             import fsspec
+            from fsspec.implementations.http import HTTPFileSystem
         except ModuleNotFoundError:
             # If filesystem is not a pyarrow filesystem and fsspec isn't
             # installed, then filesystem is neither a pyarrow filesystem nor
             # an fsspec filesystem, so we raise a TypeError.
             raise TypeError(err_msg)
```

Comment:
maybe "from None"?

Reply:
agreed, it would make error message more concise. updated.